__tests__/smoke.test.js

```diff
@@ -0,0 +1,72 @@
+const JSON5 = require('json5');
+const fs = require('fs');
+const pify = require('pify');
+const crossSpawn = require('cross-spawn');
+const rimraf = require('rimraf');
+
+async function gitCheckout(dir, repo, branch = 'master') {
```

Comment:
should we check out a specific commit hash? will the snapshots need updating whenever the branch changes as-is?

Reply:
In case of babel-preset-env we have a specific branch. For the two others I was not able to directly clone from a tag or commit, that is why I fall back to master for now. There is no other branch I could use. I am open to any suggestion making this more reliable.
But clone without `--single-branch` is taking much longer, so I decided against a full clone and checkout of a tag or commit.

Maybe a specific repo just for this purpose with all cases would be better in the long run.

Reply:
ohhh i see -- you're saying that we can't clone a specific commit so easily because we would need to first clone the repo before being able to check out the commit, and that takes too long? could we use the `-n` flag on the initial clone to prevent the initial checkout, and then run `git checkout <some-commit-hash>`?

that's my best idea for now at cloning a specific commit efficiently

otherwise, perhaps we could fork the test repos and clone the forks? 🤔

Reply:
Since this is an upgrade tool it doesn't make sense to have a test against master of a repo unless we cloned or just copy the contents as a fixture and then you wouldn't need to close it anyway?

Reply:
Just figured out that you can provide `git clone --branch` with a tag too. Would this be sufficient?
We just would need a good flowtype repo with tags and/or branches to clone since the one I use right now does not provide any of those.

@hzoo You mean we copy the files that matter for babel-upgrade into a directory below `fixtures`? We then would have to `git reset --hard` that directory after we ran babel-upgrade because we do not want to accidentally commit those changes when running it locally. That could become a problem on itself since you `Cannot do hard reset with paths.` as git phrases it.
Or we copy them from `fixtures` into a temporary directory, run the tests there and delete it afterwards like I do right now.

Reply:
yeah, i think that would be sufficient -- with the caveat that it limits us to testing repos that are using tags, i suppose

agree with @hzoo that perhaps we don't need to bother with cloning at all. maybe this is a case where the proposed "dry run" mode could be justified? https://github.com/babel/babel-upgrade/issues/6

alternatively, yes, i think copying them into a temp directory could work